### DISCUSSION TITLE
Why is the second execution of the when condition not updated, or is it the first when condition？

### DISCUSSION THREAD

**Discussion by 905174278** [spockframework/spock#1835] [Q&A] [answered]
![test1](https://github.com/spockframework/spock/assets/152140612/5b6ff1d3-b4dc-4166-814d-73ebb7a2edbb)

After (and: when:) execution：
![code1](https://github.com/spockframework/spock/assets/152140612/0a641009-9eb3-4f88-a14a-a6784f5cd0bf)

**Comment by Vampire** [comment, marked as answer]
You should probably make this as a data-driven test instead, or split it into multiple tests.
That it does not work as you expect is because you cannot override interactions like that.
The interaction from the first `when:` is still in place and is used, because it does not have any cardinality.
If you want to override previous interactions, you need to do that in `then:` like described at https://spockframework.org/spock/docs/2.3/all_in_one.html#_where_to_declare_interactions.

But as I said, you should most probably either split the test or make it data-driven anyway, then the second test can be checked, even if the first already failed, and then they are properly isolated.
